symbolic_pymc/tensorflow/printing.py

```diff
@@ -38,14 +42,62 @@ def print(self, obj):
         self.buffer.flush()
 
     def println(self, obj):
-        self.buffer.write(self.format(obj) + "\n")
-        self.buffer.flush()
+        if (
+            isinstance(self.depth_index, (list, tuple, range))
+            and self.depth_count in self.depth_index
+        ):
+            self.buffer.write(
+                self.format(obj).replace("|  " * min(self.depth_index), "... ", 1) + "\n"
+            )
+            self.buffer.flush()
+        elif self.depth_index is None:
+            self.buffer.write(self.format(obj) + "\n")
+            self.buffer.flush()
+        elif not isinstance(self.depth_index, (list, tuple, range)):
+            raise ValueError("Depth needs to be a tuple, list or range type.")
+        else:
+            pass
 
 
-def tf_dprint(obj, printer=None):
+def tf_dprint(obj, printer=None, depth_index=None):
     """Print a textual representation of a TF graph.
 
     The output roughly follows the format of `theano.printing.debugprint`.
+
+    Examples
+    --------
+    Defining concatenation in TensorFlow::
+
+        with graph_mode():
+            t1 = [[[1, 2], [2, 3]], [[4, 4], [5, 3]]]
+            t2 = [[[7, 4], [8, 4]], [[2, 10], [15, 11]]]
+            t3 = tf.concat([t1, t2], -in1)
+
+        >>> tf_dprint(t3)
+        Tensor(ConcatV2):0,	shape=[2, 2, 4]	"concat_2:0"
+        |  Op(ConcatV2)	"concat_2"
+        |  |  Tensor(Const):0,	shape=[2, 2, 2]	"concat_2/values_0:0"
+        |  |  Tensor(Const):0,	shape=[2, 2, 2]	"concat_2/values_1:0"
+        |  |  Tensor(Const):0,	shape=[]	"concat_2/axis:0"
+
+
+        >>> tf_dprint(t3, depth_index=[2])
+        |  |  Tensor(Const):0,	shape=[2, 2, 2]	"concat_2/values_0:0"
+        |  |  Tensor(Const):0,	shape=[2, 2, 2]	"concat_2/values_1:0"
+        |  |  Tensor(Const):0,	shape=[]	"concat_2/axis:0"
+
+        >>> tf_dprint(t3, depth_index=range(0, 2))
+        Tensor(ConcatV2):0,	shape=[2, 2, 4]	"concat_2:0"
+        |  Op(ConcatV2)	"concat_2"
+    
+    Parameters
+    ----------
+    obj : Tensorflow object
+        Tensorflow graph object to be represented.
+    printer : optional
+        Backend used to display the output.
+    depth_index : list, tuple or range
+        Used to index specific portions of the graph.
     """
```

Comment:
Why are you allowing non-contiguous depths?

Regardless, as I mentioned before, the interface would be much clearer and easier to use if there were optional depth upper- and lower-bound parameters only.

Reply:
Took your advice and added `depth_lower` and `depth_upper` bound parameters.